api/src/main/java/org/openmrs/module/fhir/api/client/FHIRClient.java

```diff
@@ -0,0 +1,32 @@
+package org.openmrs.module.fhir.api.client;
+
+import org.openmrs.module.fhir.api.client.dto.FHIRPatientDTO;
+import org.springframework.http.converter.HttpMessageConverter;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.Arrays;
+import java.util.Collections;
+
+
+public class FHIRClient {
+    private static final String PATIENT_CATEGORY = "patient";
+
+    private RestTemplate restTemplate = new RestTemplate();
+
+    public Object getObject(String category, String url, String username, String password) {
+        this.restTemplate.setInterceptors(Arrays.asList(new BasicAuthInterceptor(username, password),
+                new HeaderClientHttpRequestInterceptor("Accept", "application/json")));
+
+        this.restTemplate.setMessageConverters(Collections.<HttpMessageConverter<?>>singletonList(new FHIRHttpMessageConverter()));
+
+        return restTemplate.getForObject(url, resolveCategory(category));
```

Comment:
When the patient UUID is incorrect we will get the 404 error I think you should catch it and handle.